File: NSKButtonObjC/NSKButtonObjC/ViewController.m
//
//  ViewController.m
//  NSKButtonObjC
//
//  Created by NSSimpleApps on 21.08.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

#import "ViewController.h"
#import "NSKButton.h"
#import "DecodeTestViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet NSKButton *nskButton;

@property (strong, nonatomic) NSArray<NSString *> *buttonTitles;

@end

@implementation ViewController

- (void)viewDidLoad {
    
    [super viewDidLoad];
    
    NSKButton *toolBarButton1 =
    [[NSKButton alloc] initWithFrame:CGRectMake(0, 0, 40, 40)];
    toolBarButton1.nskImagePosition = NSKImagePositionTop;
    [toolBarButton1 setTitle:@"Title1" forState:UIControlStateNormal];
    [toolBarButton1 setImage:[UIImage imageNamed:@"image"] forState:UIControlStateNormal];
    [toolBarButton1 setTitleColor:[UIColor colorWithRed:0 green:122./255 blue:1 alpha:1]
                         forState:UIControlStateNormal];
    
    NSKButton *toolBarButton2 =
    [[NSKButton alloc] initWithFrame:CGRectMake(0, 0, 40, 40)];
    toolBarButton2.nskImagePosition = NSKImagePositionTop;
    [toolBarButton2 setTitle:@"Title2" forState:UIControlStateNormal];
    [toolBarButton2 setImage:[UIImage imageNamed:@"image"] forState:UIControlStateNormal];
    [toolBarButton2 setTitleColor:[UIColor colorWithRed:0 green:122./255 blue:1 alpha:1]
                         forState:UIControlStateNormal];
    
    UIBarButtonItem *buttonItem1 = [[UIBarButtonItem alloc] initWithCustomView:toolBarButton1];
    buttonItem1.width = 80;
    
    UIBarButtonItem *buttonItem2 = [[UIBarButtonItem alloc] initWithCustomView:toolBarButton2];
    buttonItem2.width = 80;
    
    UIBarButtonItem *flexibleItem =
    [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemFlexibleSpace target:nil action:nil];
    
    [self setToolbarItems:@[flexibleItem, buttonItem1, flexibleItem, buttonItem2, flexibleItem] animated:NO];
    [self.navigationController setToolbarHidden:NO animated:NO];
    
    
    self.buttonTitles = @[@"Default", @"Right", @"Top", @"Bottom"];
    
    self.nskButton.layer.borderColor = [UIColor colorWithRed:0
                                                       green:122./255
                                                        blue:1
                                                       alpha:1].CGColor;
    self.nskButton.layer.borderWidth = 1;
    [self.nskButton setTitle:self.buttonTitles.firstObject forState:UIControlStateNormal];
}


- (IBAction)changeImagePosition:(NSKButton *)sender {
    
    if (sender.nskImagePosition == NSKImagePositionBottom) {
        
        sender.nskImagePosition = NSKImagePositionDefault;
        
    } else {
        
        sender.nskImagePosition++;
    }
    
    [sender setTitle:self.buttonTitles[sender.nskImagePosition] forState:UIControlStateNormal];
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    
    if ([segue.identifier isEqualToString:@"ShowDecodeTestSegue"]) {
        
        DecodeTestViewController *decodeTestViewController = segue.destinationViewController;
        decodeTestViewController.data = [NSKeyedArchiver archivedDataWithRootObject:self.nskButton];
    }
}

@end

File: NSKButtonSwift/NSKButtonSwift/ViewController.swift
//
//  ViewController.swift
//  NSKButtonSwift
//
//  Created by NSSimpleApps on 21.08.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

import UIKit


class ViewController: UIViewController {
    @IBOutlet weak var nskButton1: NSKButton!
    @IBOutlet weak var nskButton2: NSKButton!

    fileprivate var buttonTitles = ["Default", "Right", "Top", "Bottom"]
    
    override func viewDidLoad() {
        
        super.viewDidLoad()
        
        let toolBarButton1 = NSKButton(frame: CGRect(x: 0, y: 0, width: 40, height: 40))
        toolBarButton1.nskImagePosition = .top
        toolBarButton1.setTitle("Title1", for: .normal)
        toolBarButton1.setImage(UIImage(named: "image"), for: .normal)
        toolBarButton1.setTitleColor(UIColor(red: 0, green: CGFloat(122)/255, blue: 1, alpha: 1), for: .normal)
        
        let toolBarButton2 = NSKButton(frame: CGRect(x: 0, y: 0, width: 40, height: 40))
        toolBarButton2.nskImagePosition = .top
        toolBarButton2.setTitle("Title2", for: .normal)
        toolBarButton2.setImage(UIImage(named: "image"), for: .normal)
        toolBarButton2.setTitleColor(UIColor(red: 0, green: CGFloat(122)/255, blue: 1, alpha: 1), for: .normal)
        
        let buttonItem1 = UIBarButtonItem(customView: toolBarButton1)
        buttonItem1.width = 80
        
        let buttonItem2 = UIBarButtonItem(customView: toolBarButton2)
        buttonItem2.width = 80
        
        let flexibleItem = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action:nil)
        
        self.setToolbarItems([flexibleItem, buttonItem1, flexibleItem, buttonItem2, flexibleItem], animated: false)
        self.navigationController?.setToolbarHidden(false, animated: false)
        
        self.nskButton1.layer.borderColor =
            UIColor(red: 0, green: CGFloat(122)/255, blue: 1, alpha: 1).cgColor
        self.nskButton1.layer.borderWidth = 1
        self.nskButton1.setTitle(self.buttonTitles.first, for: .normal)
        
        self.nskButton2.layer.borderColor =
            UIColor(red: 0, green: CGFloat(122)/255, blue: 1, alpha: 1).cgColor
        self.nskButton2.layer.borderWidth = 1
        self.nskButton2.setTitle(self.buttonTitles.first, for: .normal)
    }
    
    @IBAction func changeImagePosition(_ sender: NSKButton) {
        if (sender.nskImagePosition == .bottom) {
            sender.nskImagePosition = .default
        
        } else {
            let rawValue = sender.nskImagePosition.rawValue
            sender.nskImagePosition = NSKImagePosition(rawValue: rawValue + 1)!
        }
        sender.setTitle(self.buttonTitles[sender.nskImagePosition.rawValue], for: .normal)
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?)  {
    
        if segue.identifier == "ShowDecodeTestSegue" {
    
            let decodeTestViewController = segue.destination as! DecodeTestViewController
            decodeTestViewController.data = NSKeyedArchiver.archivedData(withRootObject: self.nskButton1!)
        }
    }
}

